gitpod-io/gitpod
# [JBGateway] connection-based client side additional heartbeat

## Description


Why we implement heartbeat implement on `supervisor`? Because it's hard to change token scope in JB Gateway (tech-debt) properly. Use `supervisor` to implement it is the easiest way to do so: we don't care when request failed, just need users know that this feature requires a newer installation

## Related Issue(s)

Fixes CLC-803

## How to test

**Integration tests**
Integration tests of current PR should pass

**Smoke test it doesn't break JB Gateway**
- Test via EAP Gateway
- Download matching version of Gitpod Plugin (contains branch name) in Dev channel https://plugins.jetbrains.com/plugin/18438-gitpod-gateway/versions/dev
- Install Plugin manually with matching Gateway version
- Try to create a workspace inside Gateway, it should be able to create and connect with Gateway

**Test heartbeat**
- Use `catfood.gitpod.cloud` as your Gitpod Host (in JB preferences > Tools > Gitpod > Host). You will see errors when send heartbeat as `supervisor` is not deployed yet
- Use current preview env as your Gitpod Host, there's no such error


## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-gw-hb</li>
	<li><b>🔗 URL</b> - <a href="https://hw-gw-hb.preview.gitpod-dev.com/workspaces" target="_blank">hw-gw-hb.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-gw-hb-gha.29706</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-gw-hb%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
